File: mapreduce/patent-java/src/main/java/com/epam/bigdata/mapreduce/patent/PatentCitationReducer.java
package com.epam.bigdata.mapreduce.patent;

import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Reducer;

import java.io.IOException;

public class PatentCitationReducer extends Reducer<Text, Text, Text, Text> {
    @Override
    protected void reduce(Text key, Iterable<Text> values, Context context) throws IOException, InterruptedException {
        StringBuilder builder = new StringBuilder();
        for (Text value : values) {
            builder.append(value.toString()).append(",");
        }
        builder.substring(0, builder.length() - 1);
        context.write(key, new Text(builder.toString()));
    }
}

File: mapreduce/patent-scala/src/main/scala/com/epam/bigdata/mapreduce/patent/PatentCitationReducer.scala
package com.epam.bigdata.mapreduce.patent

import java.lang.Iterable

import org.apache.hadoop.io.Text
import org.apache.hadoop.mapreduce.Reducer

import scala.collection.JavaConverters._

class PatentCitationReducer extends Reducer[Text, Text, Text, Text] {
   override def reduce(key: Text, values: Iterable[Text], context: Reducer[Text, Text, Text, Text]#Context) =
      context.write(key, new Text(values.asScala mkString (",")))
}
